packages/backend/src/tokens/factory.ts

```diff
@@ -2,14 +2,24 @@ import type { ApiClient } from '../api';
 import { API_URL, API_VERSION } from '../constants';
 import type { LoadInterstitialOptions } from './interstitial';
 import { buildPublicInterstitialUrl, loadInterstitialFromBAPI, loadInterstitialFromLocal } from './interstitial';
-import type { AuthenticateRequestOptions, AuthenticateRequestOptionsWithExperimental } from './request';
+import type { AuthenticateRequestOptions } from './request';
 import { authenticateRequest as authenticateRequestOriginal, debugRequestState } from './request';
 
 export type CreateAuthenticateRequestOptions = {
   options: Partial<
     Pick<
       AuthenticateRequestOptions,
-      'apiKey' | 'secretKey' | 'apiUrl' | 'apiVersion' | 'frontendApi' | 'publishableKey' | 'jwtKey' | 'proxyUrl'
+      | 'apiKey'
+      | 'secretKey'
+      | 'apiUrl'
+      | 'apiVersion'
+      | 'frontendApi'
+      | 'publishableKey'
+      | 'jwtKey'
+      | 'proxyUrl'
+      | 'domain'
+      | 'isSatellite'
+      | 'hasJustSynced'
```

Comment:
⛔ This function exposes the internal logic of the `authenticateRequest` function that is hard to explain to the end user. Instead of passing it as a boolean, we should replace it with a query parameter of type String or URLSearchParams that the `authenticateRequest` function will receive, do the necessary computations without exposing any of that to its public API.